src/ol/format/WFS.js

```diff
@@ -938,7 +939,11 @@ function writeProperty(node, pair, objectStack) {
         GML32.prototype.writeGeometryElement(value, pair.value, objectStack);
       }
     } else {
-      writeStringTextNode(value, pair.value);
+      if (typeof pair.value === 'string' && pair.value.search('\n') >= 0) {
```

Comment:
```suggestion
      if (typeof pair.value === 'string' && pair.value.includes('\n') >= 0) {
```
The above should work as well, and yield better performance because no regular expression is created.

Reply:
Did you mean to remove the ` >= 0` part as well, @ahocevar, like below?

```suggestion
      if (typeof pair.value === 'string' && pair.value.includes('\n')) {
```

Reply:
Yes, thanks @rdewit

Reply:
When looking at the issue more closely, I wonder if it wouldn't be better to completely skip the test and always output a CDATA block.
Indeed, it is not only end of lines that are not correctly interpreted in a text node: any white space combination (space, tab, end of line) is interpreted as a single space (as per the html/xml rules), so that for example a string with multiple consecutive spaces is also not correctly handled.
But using CDATA blocks are not the perfect solution either: they cannot contain the end of block sequence ("]]>").
They also cannot contain arbitrary binary data (it must be a valid utf8 string) but this is probably not a problem since WFS property values are supposed to be valid xml text (but I don't know if the WF specs explicitly says something about this)
For simple use cases (multi-line string, which I needed initially), CDATA blocks are working great out of the box with geoserver (and any xml parser), but maybe the correct solution for the general case would be to have an API to let the user handle the serialization himself (base64 ?)
What do you think?

Reply:
The following may be the best compromise to catch any white space combination without using CDATA blocks for single words:
[ ' ', '\n', '\t' ].some(c => pair.value.includes(c)

Reply:
Fine with me to always create a CDATA block.